File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Dossiers/DossierFolderModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Entities.Processes;
using VecompSoftware.DocSuiteWeb.Model.Processes;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Dossiers
{
    public class DossierFolderModel
    {
        #region [ Constructor ]
        public DossierFolderModel()
        {

        }
        #endregion

        #region [ Proprieties ]
        public Guid UniqueId { get; set; }

        public Guid? IdProcessFascicleTemplate { get; set; }

        public string Name { get; set; }

        public DossierFolderStatus Status { get; set; }

        public string JsonMetadata { get; set; }

        public Guid? IdFascicle { get; set; }

        public Guid? IdDossier { get; set; }

        public short? IdCategory { get; set; }

        public short? IdRole { get; set; }

        public string DossierFolderPath { get; set; }

        public short DossierFolderLevel { get; set; }
        #endregion

        #region [ Navigation Proprieties ]

        public ICollection<ProcessFascicleTemplateModel> FascicleTemplates { get; set; }

        public ICollection<DossierFolderModel> DossierFolders { get; set; }

        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Dossiers/DossierFolderModel.ts
import DossierFolderStatus = require("App/Models/Dossiers/DossierFolderStatus");
import DossierModel = require("App/Models/Dossiers/DossierModel");
import CategoryModel = require('App/Models/Commons/CategoryModel');
import FascicleModel = require("App/Models/Fascicles/FascicleModel");
import DossierFolderRoleModel = require("App/Models/Dossiers/DossierFolderRoleModel");
import ProcessFascicleTemplateModel = require("App/Models/Processes/ProcessFascicleTemplateModel");

interface DossierFolderModel {
    UniqueId: string;
    Name: string;
    Status: DossierFolderStatus;
    JsonMetadata: string;
    RegistrationDate: Date;
    RegistrationUser: string;
    LastChangedUser: string;
    LastChangedDate: Date;
    Category: CategoryModel;
    Dossier: DossierModel;
    Fascicle: FascicleModel;
    DossierFolderRoles: DossierFolderRoleModel[];
    ParentInsertId: string;
    DossierFolders: DossierFolderModel[];
    FascicleTemplates: ProcessFascicleTemplateModel[];
    DossierFolderPath: string;
    DossierFolderLevel: number;
}

export = DossierFolderModel;